core/src/main/java/org/mvcspec/ozark/servlet/OzarkServletContextListener.java

```diff
@@ -0,0 +1,91 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mvcspec.ozark.servlet;
+
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+import java.util.Set;
+import java.util.logging.Logger;
+
+import static org.mvcspec.ozark.servlet.OzarkContainerInitializer.CONTROLLER_CLASSES;
+
+/**
+ * Performs some sanity checks for controllers during startup
+ */
+@WebListener
+public class OzarkServletContextListener implements ServletContextListener {
+
+    private static final Logger log = Logger.getLogger(OzarkServletContextListener.class.getName());
+
+    private BeanManager beanManager;
+
+    @Inject
+    public OzarkServletContextListener(BeanManager beanManager) {
+        this.beanManager = beanManager;
+    }
+
+    @Override
+    public void contextInitialized(ServletContextEvent sce) {
+
+        ServletContext servletContext = sce.getServletContext();
+        Set<Class<?>> controllerClasses = (Set<Class<?>>) servletContext.getAttribute(CONTROLLER_CLASSES);
+
+        if (controllerClasses == null || controllerClasses.isEmpty()) {
+            return;
+        }
+
+        for (Class<?> controllerClass : controllerClasses) {
+            failIfNoCdiBean(controllerClass);
+            warnIfAnnotatedWithNamed(controllerClass);
+        }
+    }
+
+    /**
+     * Throws an exception if the passed controller class is not present in the CDI BeanManager.
+     * For example, this can happen with glassfish / Jersey if the application is using CDI discovery mode "annotated"
+     * and the controller class is not annotated with any scope annotation.
+     * Unlike in JAX-RS resources, MVC controllers are required to be CDI-managed beans.
+     */
+    private void failIfNoCdiBean(Class<?> controllerClass) {
+        Set<Bean<?>> controllerBeans = beanManager.getBeans(controllerClass);
+        if (controllerBeans == null || controllerBeans.isEmpty()) {
+            String message = String.format("The controller %s is not a managed CDI bean. Maybe the controller class is" +
+                    "missing a scope annotation (e.g. @RequestScope).", controllerClass.getName());
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    /**
+     * Logs a warning if the passed controller class is annotated with {@link Named}.
+     */
+    private void warnIfAnnotatedWithNamed(Class<?> controllerClass) {
+        Named namedAnnotation = controllerClass.getAnnotation(Named.class);
+        if (namedAnnotation != null) {
+            String message = String.format("Controller class %s is annotated with @Named. Typically this should not be " +
```

Comment:
Perhaps adjust the error message a bit. "Unexpected results" is actually not quite correct. Perhaps something like:

> Controller class %s is annotated with `@Named`. Typically this is not be required, because you should never access the controller directly from a view.